tests/client.py

```diff
@@ -194,6 +195,11 @@ def test__send_request(self):
         self.assertRaises(SolrError, self.solr._send_request, 'get', 'select/?q=doc&wt=json')
         self.solr.url = old_url
 
+        # Test bad core as well
+        self.solr.url = 'http://localhost:8983/solr/bad_core'
+        self.assertRaises(SolrError, self.solr._send_request, 'get', 'select/?q=doc&wt=json')
+        self.solr.url = old_url
```

Comment:
This should be a try/finally block to avoid the possibility of an unanticipated error leaving the test `self.solr.url` value in place